src/udev/net/ethtool-util.c

```diff
@@ -217,24 +217,27 @@ int ethtool_set_wol(int *fd, const char *ifname, WakeOnLan wol) {
 
 static int ethtool_get_stringset(int *fd, struct ifreq *ifr, int stringset_id, struct ethtool_gstrings **gstrings) {
         _cleanup_free_ struct ethtool_gstrings *strings = NULL;
-        struct ethtool_sset_info info =  {
-                .cmd = ETHTOOL_GSSET_INFO,
-                .reserved = 0,
-                .sset_mask = 1ULL << stringset_id,
-        };
+        struct {
+                struct ethtool_sset_info info;
+                uint32_t space;
+        } buffer;
```

Comment:
please leave the initialization right here, i.e.

```c
struct {
        struct ethtool_sset_info info;
        uint32_t space;
} buffer = {
        .info = {
                .cmd = ETHTOOL_GSSET_INFO,
                .sset_mask = UINT64_C(1) << stringset_id,
        },
};
```

(no need to initialize `.reserved` explicitly btw, the compiler automatically fills in the "gaps" with zeroes if you initialize a structure like this)